src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/RepaymentPlanContentProvider.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.cmc.claimstore.services.staff.content;
+
+import uk.gov.hmcts.cmc.claimstore.services.staff.models.RepaymentPlanContent;
+import uk.gov.hmcts.cmc.domain.models.PaymentOption;
+import uk.gov.hmcts.cmc.domain.models.RepaymentPlan;
+
+import java.time.LocalDate;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.cmc.claimstore.utils.Formatting.formatDate;
+import static uk.gov.hmcts.cmc.claimstore.utils.Formatting.formatMoney;
+import static uk.gov.hmcts.cmc.domain.models.PaymentOption.FULL_BY_SPECIFIED_DATE;
+import static uk.gov.hmcts.cmc.domain.models.PaymentOption.IMMEDIATELY;
+import static uk.gov.hmcts.cmc.domain.models.PaymentOption.INSTALMENTS;
+
+public final class RepaymentPlanContentProvider {
+
+    private RepaymentPlanContentProvider() {
+        // Utility class
+    }
+
+    public static RepaymentPlanContent create(
+        PaymentOption paymentOption,
+        RepaymentPlan repaymentPlan,
+        LocalDate payBySetDate
+    ) {
+        switch (paymentOption) {
+            case IMMEDIATELY:
+                return new RepaymentPlanContent(IMMEDIATELY.getDescription());
+            case INSTALMENTS:
+                requireNonNull(repaymentPlan, "repaymentPlan must not be null for " + INSTALMENTS);
```

Comment:
I don't think including the payment type in the message at the end gives much value as you say the type at the start too
REC